src/api/git-api.ts

```diff
@@ -40,19 +42,51 @@ export class GitExtension {
         }
     }
 
-
-    public async getRepo(showWarningIfFailed: boolean = true): Promise<Repository | undefined> {
-        const sleep = (ms: number) => new Promise(r => setTimeout(r, ms));
-        while (GitExtension.gitApi.state === 'uninitialized') { await sleep(100); }
-        const repos = GitExtension.gitApi.repositories;
-        if (repos && repos.length > 0) {
-            return repos[0];
+    public async getRepoOrShowWarning(repoSelection: RepoSelection): Promise<Repository | undefined> {
```

Comment:
@tfenster not shure about this one now. On the one hand I find it now very intuitive and readable to see `getRepoOrShowWarning(RepoSelection.choose)`, so I now that we may ask the user. On the other hand it's now always called like that which would make the RepoSelection enum obsolete. But I couldn't find a better name for now. `getOrChooseRepoOrShowWarning` seemed a bit too much somehow :D What do you think? Build it back and try to find a better name or leave it like this?

Reply:
I vote for "leave it" :)